mdagent/tools/base_tools/simulation_tools/create_simulation.py

```diff
@@ -0,0 +1,180 @@
+# a file called “md_log.txt”.
+
+simulation.reporters.append(PDBReporter('output.pdb', 1000))
+simulation.reporters.append(StateDataReporter(stdout, 1000, step=True,
+        potentialEnergy=True, temperature=True, volume=True))
+simulation.reporters.append(StateDataReporter("md_log.txt", 100, step=True,
+        potentialEnergy=True, temperature=True, volume=True))
+
+#We are using a Langevin integrator which means we are simulating in the NVT ensemble.
+# To equilibrate the temperature we just need to run the
+# simulation for a number of timesteps.
+print("Running NVT")
+simulation.step(10000)
+
+#To run our simulation in the NPT ensemble we
+# need to add in a barostat to control the pressure. We can use MonteCarloBarostat
+system.addForce(MonteCarloBarostat(1*bar, 300*kelvin))
+simulation.context.reinitialize(preserveState=True)
+
+
+print("Running NPT")
+simulation.step(10000)
+        """
+    ]
+
+    def _prompt_summary(self, query: str, llm: BaseLanguageModel = llm):
+        prompt_template = """ You're an expert programmer and in molecular dynamics.
+        Your job is to make a script to make a simmulatiuon.
+        in openmm.
+        Youre starting point is a base script that runs a protein on its own.
+        The protein itself doesnt require more preperation.
+        The forcefields, integrator, and constraints are already set up for you.
+        You need to add lines to fullfill the user requirement.
+        Your answer has to be the modified script.
+        Your answer should be a python script.
+        Dont use ''' to comment out the code, use # instead.
+        Describe your thoughts and changes before you start writing the script.
+        The script will be rum as it is, so make it completely.
+
+        The format should be as follows:
+        THOUGHTS: (Your thoughts as an openmm expert with the base script and the query)
+        CHANGES:(what modifications youre doing to the script)
+        SCRIPT: (The COMPLETE modified script)
+        FINAL THOUGHTS: (Optional, Any final thoughts or comments
+        you have about the script)
+
+
+        Base_SCRIPT:
+        {base_script}
+        Question: {query}
+        """
+
+        prompt = PromptTemplate(
+            template=prompt_template, input_variables=["base_script", "query"]
+        )
+        llm_chain = LLMChain(prompt=prompt, llm=llm)
+
+        return llm_chain.invoke(query)
+
+        # Remove leading spaces for proper formatting
+
+    def remove_leading_spaces(self, text):
+        lines = text.split("\n")
+        stripped_lines = [line.lstrip() for line in lines]
+        return "\n".join(stripped_lines)
+
+
+class ModifyScriptInput(BaseModel):
+    query: str = Field(..., description="Simmulation required by the user")
+    script: str = Field(..., description=" path to the base script file")
+
+
+class ModifyScriptTool(BaseTool):
+    name: str = "ModifyScriptTool"
+    description: str = """This tool takes a base script and a user
```

Comment:
@Jgmedina95 can you make this a bit more descriptive? I assume this is only for MD simulations, so it should be a bit more specific - might cause problems later

Reply:
You mean the tool description itself?

Reply:
done!